tests/apollo/test_skvbc_dbsnapshot.py

```diff
@@ -846,7 +846,52 @@ async def test_signed_state_snapshot_read_as_of_req_non_existent_checkpoint(self
         for ser_rsi in ser_rsis.values():
             rsi_resp = cmf_msgs.ReconfigurationResponse.deserialize(ser_rsi)[0]
             self.assertEqual(rsi_resp.response.status, cmf_msgs.SnapshotResponseStatus.SnapshotNonExistent)
-    
+
+    @with_trio
+    @with_bft_network(start_replica_cmd, selected_configs=lambda n, f, c: n == 7)
+    @verify_linearizability()
+    async def test_restore_from_snapshot_of_other(self, bft_network, tracker):
+        initial_prim = 0
+        bft_network.start_all_replicas()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network, tracker)
+        for i in range(150):  # half a window to get to a checkpoint
+            await skvbc.send_write_kv_set()
+
+        fast_paths = {}
+        for r in bft_network.all_replicas():
+            nb_fast_path = await bft_network.get_metric(r, bft_network, "Counters", "totalFastPaths")
+            fast_paths[r] = nb_fast_path
+
+        crashed_replica = list(bft_network.random_set_of_replicas(1, {initial_prim}))[0]
+        bft_network.stop_replica(crashed_replica)
+        for i in range(150):  # run till the next checkpoint
+            await skvbc.send_write_kv_set()
+
+        src_replica = list(bft_network.random_set_of_replicas(1, without={crashed_replica}))
+        await self.wait_for_stable_checkpoint(bft_network, src_replica, 300)
+        await self.wait_for_created_snapshots_metric(bft_network, src_replica, 2)
+        snapshot_id = await bft_network.get_metric(src_replica[0], bft_network,
```

Comment:
This metric is used by many tests, you can extract a function for it

Reply:
done